### DISCUSSION TITLE
Strict mode for rendering and optionally preview

### DISCUSSION THREAD

**Discussion by cpcloud** [quarto-dev/quarto-cli#6677] [Feature Requests]
### Description

I would love a way to fail `quarto render` and optionally `quarto preview` if any warnings are generated. This is especially useful when deploying documentation, as I would like to prevent anything from coming through that's not going to render as intended, refer to documents that don't exist, etc.

**Comment by mcanouil** [comment]
@cscheid @jjallaire Is it something that might already exist but is not documented or something that could be added?

**Reply by cscheid** [reply]
Not that I'm aware of; our test suite has some code to check for no warnings, but we don't have something that's comprehensive through the whole pipeline. We should, but it's a large project to do it.

**Comment by skwde** [comment]
I would love to have that as well.

Is this somewhat on the roadmap?

This is a feature, e.g. also present in `mkdocs build`, they basically use a counter of the appearing `LOG.warning`'s and fail if non zero, the source is here: https://github.com/mkdocs/mkdocs/blob/master/mkdocs/commands/build.py#L249

At least one of my use cases would be to fail if a certain `meta` key is not defined.
Currently this only triggers a warning like

```out
WARNING (/opt/quarto/share/filters/main.lua:17878) Unknown meta key <key> specified in a metadata Shortcode.
```

**Comment by skwde** [comment]
@mcanouil  @cscheid Is something like this included in quarto 2?

I keep getting back at this.
